File: kickWorker.js
const dayjs = require("dayjs");

const APPS_SCRIPT_URL = process.env.APPS_SCRIPT_URL;
const GROUP_ID = process.env.GROUP_ID;

async function getData() {
  const res = await fetch(APPS_SCRIPT_URL + "?action=get");
  return await res.json();
}

async function checkKick(bot) {
  try {
    const data = await getData();
    const today = dayjs().format("YYYY-MM-DD");

    for (let row of data) {
      if (row.kickDate === today && row.out !== "✔") {
        try {
          await bot.telegram.banChatMember(GROUP_ID, row.userId);

          await bot.telegram.sendMessage(
            row.userId,
`Langganan Kamu Telah Selesai ...
Silahkan Upgrade Langganan kamu Dengan Disini https://t.me/AITOOLSIGNAL_BOT?start`
          );

          console.log("KICK:", row.userId);
        } catch (e) {
          console.log("ERR:", e.message);
        }
      }
    }
  } catch (err) {
    console.log(err.message);
  }
}

module.exports = { checkKick };

File: kick_worker.py
import asyncio
import requests

async def kick_worker_async(app):

    while True:
        try:
            r = requests.get(APPS_SCRIPT_URL, timeout=30)
            data = r.json().get("data", [])

            print("━━━━━━━━━━━━━━")
            print("CHECKING KICK DATA...")

            for u in data:
                user_id = u.get("userId")
                kick_date = u.get("kickDate")
                out = u.get("out")

                print("CHECK:", user_id, kick_date, out)

                if not user_id:
                    continue

                if out == "✔":
                    continue

                if is_expired(kick_date):

                    try:
                        print("🔥 TRY KICK:", user_id)

                        # 🔥 IMPORTANT: pakai app.bot (BUKAN Bot() baru)
                        await app.bot.ban_chat_member(
                            chat_id=TARGET_GROUP,
                            user_id=int(user_id)
                        )

                        await app.bot.send_message(
                            chat_id=int(user_id),
                            text=f"❌ Membership kamu habis.\n👉 Start lagi: {REDIRECT_LINK}"
                        )

                        requests.post(APPS_SCRIPT_URL, json={
                            "action": "markOut",
                            "userId": user_id
                        }, timeout=10)

                        print("🔥 KICKED:", user_id)

                    except Exception as e:
                        print("❌ KICK ERROR:", e)

        except Exception as e:
            print("❌ KICK LOOP ERROR:", e)

        await asyncio.sleep(60)
